src/app/api/users/route.ts

```diff
@@ -0,0 +1,30 @@
+import { user } from "@/dao/user.dao";
+import { NextResponse, NextRequest } from "next/server";
+import { z } from "zod";
+
+const checkGetAllUsersParams = z.object({
+  username: z.string().optional(),
+  email: z.string().optional(),
+  page: z.number().optional().transform((val) => (val ? Number(val) : undefined)),
+  pageSize: z.number().optional().transform((val) => (val ? Number(val) : undefined)),
```

Comment:
Let's simplify this by using zod default

```ts
const DEFAULT_PAGE = 1;
const DEFAULT_PAGE_SIZE = 10;
```
```suggestion
  page: z.number().default(DEFAULT_PAGE),
  pageSize: z.number().default(DEFAULT_PAGE_SIZE),
```

- https://zod.dev/?id=default